2.1.8 breaks projects using Jetbrains Annotations

A quick reproducer:

```
#!/usr/bin/env bash
set -euo pipefail

cat > .bazelrc <<'EOF'
build --java_language_version=17
build --java_runtime_version=remotejdk_17
build --tool_java_language_version=17
build --tool_java_runtime_version=remotejdk_17
EOF

cat > MODULE.bazel <<'EOF'
bazel_dep(name = "rules_kotlin", version = "2.1.8") # with 2.1.7 it works

http_file = use_repo_rule("@bazel_tools//tools/build_defs/repo:http.bzl", "http_file")

http_file(
    name = "annotations",
    downloaded_file_path = "annotations-26.0.2.jar",
    sha256 = "2037be378980d3ba9333e97955f3b2cde392aa124d04ca73ce2eee6657199297",
    url = "https://cache-redirector.jetbrains.com/repo1.maven.org/maven2/org/jetbrains/annotations/26.0.2/annotations-26.0.2.jar",
)
EOF

cat > BUILD <<'EOF'
load("@rules_kotlin//kotlin:jvm.bzl", "kt_jvm_import",  "kt_jvm_library")

kt_jvm_import(
    name = "annotations",
    jar = "@annotations//file",
    visibility = ["//visibility:public"],
)

kt_jvm_library(
    name = "Foo",
    srcs = ["Foo.java"],
    deps = [":annotations"],
)
EOF

cat > Foo.java <<'EOF'
import org.jetbrains.annotations.NotNull;

class Foo {
    Object @NotNull [] test = new Object[0];
}
EOF

bazel build ...
```
It's gonna fail

```
ERROR: /home/andrzej.gluszak/code/junk/blah/test/BUILD:9:15: Building Foo-java.jar (1 source file) failed: (Exit 1): java failed: error executing Javac command (from target //:Foo) external/rules_java++toolchains+remotejdk21_linux/bin/java '--add-exports=jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED' '--add-exports=jdk.compiler/com.sun.tools.javac.main=ALL-UNNAMED' ... (remaining 19 arguments skipped)
Foo.java:4: error: annotation @NotNull not applicable in this type context
    Object @NotNull [] test = new Object[0];
           ^
Target //:Foo failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 15.538s, Critical Path: 2.23s
INFO: 11 processes: 6 internal, 4 linux-sandbox, 1 worker.
ERROR: Build did NOT complete successfully
```

Then downgrade to 2.1.7 and it will work :)

I suspect it's because of this commit https://github.com/bazelbuild/rules_kotlin/commit/43be56182786764ef0a5d1d6ea00dce3c198213c

The order of deps has been changed:

was `dep_infos = deps_java_infos + [_java_info(d) for d in deps] + [toolchains.kt.jvm_stdlibs]` and now is `dep_infos = [toolchains.kt.jvm_stdlibs] + associates.dep_infos + deps_java_infos +  [_java_info(d) for d in deps]`.

I guess now it picks up `org.jetbrains.annotations.NotNull` from the stdlib and it probably has different semantics or simply is an older version?

The fix is probably to restore previous ordering, but I haven't verified it yet.

--- a/kotlin/internal/jvm/jvm_deps.bzl
+++ b/kotlin/internal/jvm/jvm_deps.bzl
@@ -29,10 +29,10 @@ def _jvm_deps(ctx, toolchains, associate_deps, deps = [], deps_java_infos = [], 
         associates = associate_deps,
     )
     dep_infos = (
-        [toolchains.kt.jvm_stdlibs] +
-        associates.dep_infos +
         deps_java_infos +
-        [_java_info(d) for d in deps]
+        [_java_info(d) for d in deps] +
+        associates.dep_infos +
+        [toolchains.kt.jvm_stdlibs]
     )
 
     # Reduced classpath, exclude transitive deps from compilation

--- a/src/test/starlark/internal/jvm/jvm_deps_tests.bzl
+++ b/src/test/starlark/internal/jvm/jvm_deps_tests.bzl
@@ -266,6 +266,113 @@ def _transitive_from_exports_test(name):
 def _transitive_from_associates_test(name):
     _abi_test(name, _transitive_from_associates_test_impl)
 
+def _dep_infos_ordering_test_impl(env, target):
+    """Test that user deps take precedence over stdlib in dep_infos ordering.
+
+    This is a regression test for https://github.com/bazelbuild/rules_kotlin/issues/1368
+    where the stdlib was placed first in the classpath, causing it to shadow user
+    dependencies with conflicting classes (e.g., org.jetbrains.annotations.NotNull).
+    """
+    user_dep_java_info = JavaInfo(
+        compile_jar = _file(env.ctx.attr.user_dep_jar),
+        output_jar = _file(env.ctx.attr.user_dep_jar),
+    )
+
+    associate_java_info = JavaInfo(
+        compile_jar = _file(env.ctx.attr.associate_jar),
+        output_jar = _file(env.ctx.attr.associate_jar),
+    )
+
+    stdlib_java_info = JavaInfo(
+        compile_jar = _file(env.ctx.attr.jvm_jar),
+        output_jar = _file(env.ctx.attr.jvm_jar),
+    )
+
+    associate_deps = [
+        {
+            JavaInfo: associate_java_info,
+            _KtJvmInfo: _KtJvmInfo(
+                module_name = "associate_name",
+            ),
+        },
+    ]
+
+    user_deps = [
+        {
+            JavaInfo: user_dep_java_info,
+        },
+    ]
+
+    fake_ctx = struct(
+        label = target.label,
+        attr = struct(
+            module_name = "",
+            tags = [],
+        ),
+    )
+
+    toolchains = struct(
+        kt = struct(
+            experimental_remove_private_classes_in_abi_jars = False,
+            experimental_prune_transitive_deps = False,
+            experimental_strict_associate_dependencies = False,
+            jvm_stdlibs = stdlib_java_info,
+        ),
+    )
+
+    result = _jvm_deps_utils.jvm_deps(
+        ctx = fake_ctx,
+        toolchains = toolchains,
+        associate_deps = associate_deps,
+        deps = user_deps,
+    )
+
+    # Assert the ordering: result.deps should be [user_deps, associates, stdlib]
+    env.expect.that_int(len(result.deps)).equals(3)
+
+    # Verify compile_jars contains all deps in correct order
+    classpath_list = result.compile_jars.to_list()
+    user_dep_file = _file(env.ctx.attr.user_dep_jar)
+    stdlib_file = _file(env.ctx.attr.jvm_jar)
+
+    # Find indices of user dep and stdlib in classpath
+    user_dep_idx = -1
+    stdlib_idx = -1
+    for idx, jar in enumerate(classpath_list):
+        if jar == user_dep_file:
+            user_dep_idx = idx
+        if jar == stdlib_file:
+            stdlib_idx = idx
+
+    # Both should be found
+    env.expect.that_bool(user_dep_idx >= 0).equals(True)
+    env.expect.that_bool(stdlib_idx >= 0).equals(True)
+
+    # User dep should come before stdlib in the classpath
+    env.expect.that_bool(user_dep_idx < stdlib_idx).equals(True)
+
+def _dep_infos_ordering_test(name):
+    util.helper_target(
+        native.filegroup,
+        name = name + "_subject",
+        srcs = [],
+    )
+    analysis_test(
+        name = name,
+        impl = _dep_infos_ordering_test_impl,
+        target = name + "_subject",
+        attr_values = {
+            "user_dep_jar": util.empty_file(name + "user_dep.jar"),
+            "associate_jar": util.empty_file(name + "associate.jar"),
+            "jvm_jar": util.empty_file(name + "jvm.jar"),
+        },
+        attrs = {
+            "user_dep_jar": attr.label(allow_files = True),
+            "associate_jar": attr.label(allow_files = True),
+            "jvm_jar": attr.label(allow_files = True),
+        },
+    )
+
 def jvm_deps_test_suite(name):
     test_suite(
         name,
@@ -274,5 +381,6 @@ def jvm_deps_test_suite(name):
             _fat_abi_test,
             _transitive_from_exports_test,
             _transitive_from_associates_test,
+            _dep_infos_ordering_test,
         ],
     )

